x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/stats/Metrics.java

```diff
@@ -36,8 +39,24 @@ public String toString() {
     private final Map<QueryMetric, Map<OperationType, CounterMetric>> opsByTypeMetrics;
     // map that holds one counter per esql query "feature" (eval, sort, limit, where....)
     private final Map<FeatureMetric, CounterMetric> featuresMetrics;
+    private final Map<String, CounterMetric> functionMetrics;
     protected static String QPREFIX = "queries.";
     protected static String FPREFIX = "features.";
+    protected static String FUNC_PREFIX = "functions.";
+
+    private static final EsqlFunctionRegistry funcitonRegistry = new EsqlFunctionRegistry().snapshotRegistry();
```

Comment:
Typo: `funciton` -> `function`
Also, you don't need another function registry instance here. `Metrics` is created in PlanExecutor, which already has access to a function registry instance.